rad/prelude/lens.rad

```diff
@@ -107,6 +107,24 @@
                  {:a {:b {:c 0}}})
            {:a {:b {:c 1}}})
 
+;; @nth
+
+(def @nth
+  "Lenses into the nth element of a vector"
+  (fn [n]
+    (make-lens
+      (fn [s] (nth n s))
+      (fn [val s]
+         (<> (take n s) (add-left val (drop (+ n 1) s)))))))
```

Comment:
Should we use a primitive for this based on `Seq.update`

Reply:
Probably. Ultimately I think it's more important for this to work on both vectors and lists than to be performant (right now, at least). But we can of course still have a vec/list `update` primitive. 

#196 is also relevant. 

OK if I add an issue for this?